src/status_im2/contexts/chat/messages/content/text/view.cljs

```diff
@@ -73,10 +74,25 @@
     blocks))
 
 (defn render-parsed-text
-  [{:keys [content]}]
-  (reduce render-block
-          [:<>]
-          (:parsed-text content)))
+  [{:keys [content edited-at]}]
+  (let [parsed-text-count           (when edited-at
+                                      (count (:parsed-text content)))
+        parsed-text-with-edited-tag (when edited-at
+                                      (update (:parsed-text content)
```

Comment:
I think we should explore in a future work using a subscription for pre-processing `message-data`. Only once the data is ready to be transformed to hiccup the view components should get into the action. This can help with performance. Right now `render-parsed-text` is a bit heavy on logic.

I've noticed every time I send any text, single char or more, the top-level view component `text-content` and all its children render 3 times. Something we should probably review if we want to improve even further the performance.

Reply:
I kept adding checks to prevent the calculation of the edited message body on non edited messages

Reply:
But you're correct .. We need to somehow cache the result of the message rendering computation for subsequent renders